src/Entity/OgRole.php

```diff
@@ -177,8 +177,39 @@ public function setRoleType($role_type) {
   /**
    * {@inheritdoc}
    */
+  public function getName() {
+    // If the name is not set yet, try to derive it from the ID.
+    if (empty($this->name) && !empty($this->id())) {
+      list(, , $name) = explode('-', $this->id());
```

Comment:
This seems slightly limited. Maybe it would be better to explode, but pop off the last element and use that?